File: python/LeastCommonMultiple.py

from random import * 

def naive(a, b):
	for l in range(1, a * b + 1):
		if l % a == 0 and l % b == 0:
			return l
	return a * b

def fast(a, b):
	return a * b // GCD(a, b)

def GCD(a, b):
	if b == 0:
		return a
	return GCD(b, a % b)

a, b = map(int, input().split())
# print(naive(a, b))
print(fast(a, b))

'''
n = int(input())

while 1:
	a = randint(1, n)
	b = randint(1, n)
	
	if naive(a, b) == fast(a, b):
		print(a, b, " - ok")
	else:
		print(a, b, " - fail")
		break
'''

File: java/src/LeastCommonMultiple.java
import java.util.Random;
import java.util.Scanner;

public class LeastCommonMultiple {

	private static long naive(int a, int b) {
		for (long i = 1 ; i < (long) a * b + 1 ; i++) {
			if (i % a == 0 && i % b == 0) {
				return i;
			}
		}
		
		return (long) a * b;
	}
	
	private static long fast(int a, int b) {
		return (long) a * b / GCD(a, b);
	}
	
	private static int GCD(int a, int b) {
		if (b == 0)
			return a;

		return GCD(b, a % b);
	}
	
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int a = scanner.nextInt();
		int b = scanner.nextInt();
		
		// System.out.println(naive(a, b));
		System.out.println(fast(a, b));
		scanner.close();
	}
	
	/*public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int n = scanner.nextInt();
		
		Random rand = new Random();
		
		while(true) {
			int a = rand.nextInt(n) + 1;
			int b = rand.nextInt(n) + 1;
			System.out.print(a + "    " + b);

			if (naive(a, b) == fast(a, b)){
				System.out.println(" - ok");
			} else {
				System.out.println(" - fail");
				break;
			}
		}

		scanner.close();
	}*/
}
